readthedocs/proxito/views/hosting.py

```diff
@@ -23,37 +55,120 @@ def get(self, request):
                 status=400,
             )
 
+        client_version = request.headers.get("X-RTD-Hosting-Integrations-Version")
+        if not client_version:
+            return JsonResponse(
+                {
+                    "error": ClientError.VERSION_HEADER_MISSING,
+                },
+                status=400,
+            )
+        try:
+            client_version = packaging.version.parse(client_version)
+            if client_version.major not in CLIENT_VERSIONS_SUPPORTED:
+                raise ClientError
+        except packaging.version.InvalidVersion:
+            return JsonResponse(
+                {
+                    "error": ClientError.VERSION_INVALID,
+                },
+                status=400,
+            )
+        except ClientError:
+            return JsonResponse(
+                {"error": ClientError.VERSION_NOT_CURRENTLY_SUPPORTED},
+                status=400,
+            )
+
         unresolved_domain = request.unresolved_domain
         project = unresolved_domain.project
 
         unresolved_url = unresolver.unresolve_url(url)
         version = unresolved_url.version
+        filename = unresolved_url.filename
 
         project.get_default_version()
         build = version.builds.last()
 
-        # TODO: define how it will be the exact JSON object returned here
-        # NOTE: we could use the APIv3 serializers for some of these objects
-        # if we want to keep consistency. However, those may require some
-        # extra db calls that we probably want to avoid.
+        data = ClientResponse().get(client_version, project, version, build, filename)
+        return JsonResponse(data, json_dumps_params=dict(indent=4))
+
+
+class NoLinksMixin:
+
+    """Mixin to remove conflicting fields from serializers."""
+
+    FIELDS_TO_REMOVE = (
+        "_links",
+        "urls",
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(NoLinksMixin, self).__init__(*args, **kwargs)
+
+        for field in self.FIELDS_TO_REMOVE:
+            if field in self.fields:
+                del self.fields[field]
+
+            if field in self.Meta.fields:
+                del self.Meta.fields[self.Meta.fields.index(field)]
+
+
+# NOTE: the following serializers are required only to remove some fields we
+# can't expose yet in this API endpoint because it running under El Proxito
+# which cannot resolve some dashboard URLs because they are not defined on El
+# Proxito.
+#
+# See https://github.com/readthedocs/readthedocs-ops/issues/1323
+class ProjectSerializerNoLinks(NoLinksMixin, ProjectSerializer):
+    pass
+
+
+class VersionSerializerNoLinks(NoLinksMixin, VersionSerializer):
+    pass
+
+
+class BuildSerializerNoLinks(NoLinksMixin, BuildSerializer):
+    pass
+
+
+class ClientResponse:
```

Comment:
Do we want to do this?

```suggestion
class AddonsResponse:
```

Reply:
I rename everything called "client" to "addons" 👍🏼